src/imitation/algorithms/bc.py

```diff
@@ -155,7 +161,8 @@ def _build_tf_graph(self):
             )
 
             _l2_norms = [
-                tf.reduce_mean(tf.norm(w)) for w in model_scope.trainable_variables()
+                tf.reduce_mean(tf.nn.l2_loss(w))
+                for w in model_scope.trainable_variables()
             ]
             self._l2_norm = tf.reduce_sum(_l2_norms)
```

Comment:
~Maybe replace this with `reduce_mean` for consistency with the `reduce_mean` on line 164?~
**Edit:** actually, it would make more sense to replace the `reduce_mean` on line 164 with a `reduce_sum`. That's at least consistent with the `$\lambda \|\theta\|_2^2$` notation that people use in papers.

Reply:
Double reduce sum makes sense

Reply:
Ugh, actually I'm thinking maybe we should keep the mean on the inside, but only mean over the batch dimension?

If we use double reduce sum, then the L2 Loss scales with the batch size, which seems wrong, or at least annoying from a Tensorboard and hyperparam tuning perspective.

Reply:
These are just weights, right? There shouldn't be a batch dimension.

Reply:
Oh, there isn't a batch index.

Reply:
As it turns out, l2_loss returns a scalar anyways, so a reduction is not necessary inside the list comprehension.